src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,48 +1,77 @@
-import React from 'react';
+import { useEffect, useState, FC } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  setModalTodoHandler: (v:Todo | undefined) => void,
+  modalTodo?: Todo,
+};
+
+export const TodoModal: FC<Props> = ({
+  setModalTodoHandler, modalTodo,
+}) => {
+  const [todoUser, setTodoUser] = useState<User>();
+
+  useEffect(() => {
+    if (modalTodo) {
+      getUser(modalTodo.userId).then(data => {
+        setTodoUser(data);
+      });
+    }
+  }, [modalTodo]);
+
+  const closeModalHandler = () => {
+    setModalTodoHandler(undefined);
+  };
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
-        <Loader />
-      ) : (
+      {modalTodo && todoUser ? (
         <div className="modal-card">
           <header className="modal-card-head">
             <div
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              Todo #
+              {modalTodo.id}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
               type="button"
               className="delete"
               data-cy="modal-close"
+              onClick={closeModalHandler}
             />
           </header>
 
           <div className="modal-card-body">
             <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
+              {modalTodo?.title}
             </p>
 
             <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
+              {modalTodo.completed ? (
+                <strong className="has-text-success">Done</strong>
+              ) : (
+                <strong className="has-text-danger">Planned</strong>
+              )}
 
               {' by '}
 
-              <a href="mailto:Sincere@april.biz">
-                Leanne Graham
+              <a href={todoUser?.email}>
+                {todoUser?.name}
```

Comment:
```suggestion
              <a href={todoUser.email}>
                {todoUser.name}
```
you render this information only when  `modalTodo && todoUser` exists